db/db_impl_write.cc

```diff
@@ -262,6 +298,15 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
     mutex_.Lock();
     MarkLogsSynced(logfile_number_, need_log_dir_sync, status);
     mutex_.Unlock();
+    // Requesting sync with concurrent_prepare_ is expected to be very rare. We
+    // hance provide a simple implementation that is not necessarily efficient.
+    if (concurrent_prepare_) {
+      if (manual_wal_flush_) {
+        status = FlushWAL(true);
+      } else {
+        status = SyncWAL();
```

Comment:
If we do not do call MarkLogsSynced here then SyncWAL will be blocked on this:
```
 645     while (logs_.front().number <= current_log_number &&
 646            logs_.front().getting_synced) {
 647       log_sync_cv_.Wait();
 648     }
```
I know it is inefficient but we favored simplicity here since i) it is expected to be rare, ii) this code is going to be retired in a few month after our next generation of transactions.